routes/user.js

```diff
@@ -0,0 +1,55 @@
+const express = require('express');
+const router = express.Router();
+const crypt = require('bcryptjs')
+const restricted = require('../data/restricted-middleware');
+
+const db = require('../data/dbConfig.js');
+const Users = require('../data/users_model.js')
+
+const errors = { 
+  '19':'Another entree has the same value no duplicates'
+}
+
+router.post('/register',(req,res)=>{
+  let user = req.body;
+  const hash = crypt.hashSync( req.body.password);
+  user.password=hash;
+  Users.add(user)
+    .then(saved => {
+      res.status(201).json(saved);
+    })
+      .catch(error => {res.status(500).json(error)});
+});
+
+router.post('/login', (req, res) => {
+  let { user, password } = req.body;
+
+  Users.findBy({ user })
+    .first()
+    .then(user => {  
+      if (user && crypt.compareSync(password,user.password)) {
+        req.session.user = user;
+        res.status(200).json({ message: `Welcome ${user.user}!, have a cookie` });
+      } else {
+        res.status(401).json({ message: 'Invalid Credentials' });
+      }
+    })
+    .catch(error => {
+      res.status(500).json(error);
+    });
+});
+
+router.get('/', restricted, (req, res) => {
+  Users.find()
+    
+    .then(user => {
+      res.json(user);
+    })
+    .catch(err => res.send(err));
+});
+
+
+// const hash = bcrypt.hashSync( , );
```

Comment:
Don't forget to remove unnecessary comments--unless this is a note for yourself, in which case I'd suggest rewriting it to be useful for other developers reading your code, too!